crates/wit-component/src/dummy.rs

```diff
@@ -192,13 +192,13 @@ fn push_imported_future_and_stream_intrinsics(
             TypeDefKind::Future(_) => {
                 wat.push_str(&format!(
                     r#"
-(import {module:?} "[future-new-{i}]{name}" (func (result i32)))
+(import {module:?} "[future-new-{i}]{name}" (func (result i64)))
 (import {module:?} "[future-read-{i}]{name}" (func (param i32 i32) (result i32)))
 (import {module:?} "[future-write-{i}]{name}" (func (param i32 i32) (result i32)))
 (import {module:?} "[future-cancel-read-{i}]{name}" (func (param i32) (result i32)))
 (import {module:?} "[future-cancel-write-{i}]{name}" (func (param i32) (result i32)))
-(import {module:?} "[future-close-readable-{i}]{name}" (func (param i32 i32)))
-(import {module:?} "[future-close-writable-{i}]{name}" (func (param i32 i32)))
+(import {module:?} "[future-close-readable-{i}]{name}" (func (param i32)))
+(import {module:?} "[future-close-writable-{i}]{name}" (func (param i32)))
```

Comment:
This is a basic summary of the changes here, notably:

* `{future,stream}.new` intrinsics return a two 32-bit handles as a result packed into a 64-bit integer. It kind of sucks we don't have multi-value and I also didn't want to go the route of adding canonopts to these intrinsics just to write values into memory. It's a bit wonky since bit packing doesn't happen anywhere else in the canonical ABI but it happens in other async intrinsics so maybe ok?
* For `{future,stream}.close-{readable,writable}` the only change is to drop the error-context parameter.